Cython/Compiler/ExprNodes.py

```diff
@@ -7377,6 +7405,13 @@ def analyse_types(self, env, skip_children=False):
         # not setting self.type here, subtypes do this
         return self
 
+    def analyse_as_possible_annotation_type(self, pos, env):
+        # work on a copy of self
+        self = copy.copy(self)
+        for i, arg in enumerate(self.args):
+            self.args[i] = arg.analyse_as_possible_annotation_type(pos, env)
```

Comment:
You only copied `self`, not `self.args`. Maybe you wanted the loop to be more like
`self.args = [arg.analyse…() for arg in self.args]` ?